README
README for Open-ETickets 

Open-ETickets is a fork of Fusion Ticket. Most of this code is taken from there. Due to
disagreements over the intepretation of the GPL with the current lead dev, I have forked
it and moved to GitHub.

There is NO SUPPORT for this. If you wish to use a supported project, I strongly recommend
you use the original.

--Rob Thomas - 2012-02-08 - xrobau@gmail.com

Original README below:

Welcome to Fusion Ticket: online ticket reservation system for festivals, theaters 
or just your party!

Fusion Ticket uses the GNU v3 Licence by installing this software you are agreeing to
the terms of that licence.

Installation instructions can be found in INSTALL file.

For any additionnal information, feedback, bug reports and other please see:

http://www.fusionticket.co.uk/


Current Developers:-
	2007-2008
	Christopher Jenkins A.K.A. CJ
	info@fusionticket.co.uk
	

Original Developers:-
	2003-2005
	Anna + Stani = webtigo.ch
	Ramosch, Switzerland, 
	info@phpmyticket.com


changelog.txt
Change Log
/**
%%%copyright%%%
 *
 * FusionTicket - ticket reservation system
 *  Copyright (C) 2007-2011 Christopher Jenkins. All rights reserved.
 *
 * Original Design:
 *	phpMyTicket - ticket reservation system
 * 	Copyright (C) 2004-2005 Anna Putrino, Stanislav Chachkov. All rights reserved.
 *
 * This file is part of fusionTicket.
 *
 * This file may be distributed and/or modified under the terms of the
 * "GNU General Public License" version 3 as published by the Free
 * Software Foundation and appearing in the file LICENSE included in
 * the packaging of this file.
 *
 * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING
 * THE WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE.
 *
 *
 * The "GNU General Public License" (GPL) is available at
 * http://www.gnu.org/copyleft/gpl.html.
 *
 * Contact info@noctem.co.uk if any conditions of this licencing isn't
 * clear to you.
 */

////////////////////////////////////
// THIS FILE IS NOW DEPRICATED
// IT IS ONLY KEEP FOR HISTORIC REASONS
// FOR bug/feature tracking use http://mantis.fusionticket.org
////////////////////////////////////


Feedback:
	- When users signup add in another email feild to check there email is the same as the first field.
//TODO: FIX templates
//TODO: PARTIAL - Let users change there details.
//BUG: Some links in the admin page don't work.
//BUG: Only Cash and paypal orders work in POS?
//BUG: Paid Sent has no colour in admin area.
TESTING - Let users see current orders, and past orders (allow to cancel change etc).

-- BETA 3 PATCH 1 ---
06/06/2008
Ver 1.3.5 Build 45
- Fixed PDF's header error
- Removed White space which could cause potential problems.
- Fixed problem in PoS where it would not create a new user if you didnt click "New User"
- Fixed Process Order Link


-- BETA 3 - PUBLIC --
2/06/2008
Ver 1.3.5 Build 44
- Cleaned up templates
- New standard template! Thank you to Paul AKA pangea
- Global Email is taken from Organiser Settings now (activation page etc)
- Currency is taken from organiser settings (again)
- Fixed some object errors
- Improved User Login, cleaned up the code and improved security
- Increased state feild size in DB.



24/05/2008
Ver 1.3.4 Build 43
- Added Processing Section in POS for manual system intergration and easy order processing. (too be expanded at a later date)
- Login will redirect to last page when logging on.
- Updated Footer with correct link



20/05/2008
Ver 1.3.4 Build 42
- Added max res orders so users cant go an reserve the whole dam show any pay nothing!
	It is very much recommend if you use Reservation that this feature is enabled it is enabled by default!
- Improved transaction code in class.shopdb.php Fixed some strange happenings, Transactions are near 100% reliable now.
- Users can now change the size of a event mid show only applicable to unnumbered seats - Utilised Exsisting Code.
	Event Must be unpubished (nautraly).
- Shows total tickets bought in catview admin area



28/04/2008
Ver 1.3.4 Build 41
- Updated htmlMimeMail to v5 designed for php5
- Updated PEAR Core system, Its not used as far as im aware but updated it to php5 version just incase.
- Added resend activation email
- Added Extra lang items in lang file need to be added to the rest of the languages
- Added Extra db fields, SQL in 'Current Patch' folder
- Added Extra options to Config in Admin Area


--BETA 2 --

27/04/2008
Ver 1.3.4 Build 40
- Fixed MySQLi Error in class.shopdb.php should display the error properly and compile pages fine now
- Fixed missing colloums
- Fixed Missing data for ordering tickets.
- Changed quote var to mysqli over mysql in Handlings.php
- Updated Smarty Version
NOTES: Uses PHP send mail the script will only send mail from the domain that its at! Check Mail Security Settings!

--BETA 1 -- Still late alpha i wanted a beta version for some feedback so here it is.

14/04/2008
Ver 1.3.3 Build 39
- Fixed no date bug in order template
- Removed SpryAssests should show basket
- Tested site with cookies off, seem so be no problems unless logging in to the admin site.


--PRE ALPHA 1.3 BUILD--
31/03/2008
Ver 1.3.2 Build 38
- Added Order Options in the admin area so auto shop settings can be changed. Apears to work :)
- Added Reserved Labling on order handlings
- Updated Missing lanuage text in admin file.

12/03/2008
Ver 1.3.2 Build 37
- Can now select exsisting users when ordering/reserving in the box office
- Added new field in users, which counts the users total orders, when ever a new order is made the count is increased.
this gives a more realistic veiw, when old orders are deleated you can still still you popular customers.

11/03/2008
Ver 1.3.2 Build 36
- Now finds earlyest event date for ordering instead of see weather it alt date. adds more functality.
- Handling and reserve buttons check event date agaisnt shop config offset to see if they should be displayed.
- Cant reserve length of reserve time+2days before an event as you would have a reservation overlapping an event which means un paid tickets. I will make it so reservations before the event can be changed.



11/03/2008
Ver 1.3.2 Build 35
- Orders will now follow alt methods, Handlings will either check the cart or event_date input, if no input the use_alt lines will be ignored so payments arent asked if they are alt payments are not as we dont need to ask the system, it knows the handling id.
- handling will just check if an alt is slected not which one. Which works just not how i want it yet.
- created update.php, doesnt require smarty for the class.
- created tickets function, will grab the min date from a list of tickets, so when ordering stuff the min date is allways picked out,



09/03/2008
Ver 1.3.2 Build 34

nospam.php
<?php
/**
%%%copyright%%%
 *
 * FusionTicket - ticket reservation system
 *  Copyright (C) 2007-2011 Christopher Jenkins, Niels, Lou. All rights reserved.
 *
 * Original Design:
 *	phpMyTicket - ticket reservation system
 * 	Copyright (C) 2004-2005 Anna Putrino, Stanislav Chachkov. All rights reserved.
 *
 * This file is part of FusionTicket.
 *
 * This file may be distributed and/or modified under the terms of the
 * "GNU General Public License" version 3 as published by the Free
 * Software Foundation and appearing in the file LICENSE included in
 * the packaging of this file.
 *
 * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING
 * THE WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR
 * PURPOSE.
 *
 * Any links or references to Fusion Ticket must be left in under our licensing agreement.
 *
 * By USING this file you are agreeing to the above terms of use. REMOVING this licence does NOT
 * remove your obligation to the terms of use.
 *
 * The "GNU General Public License" (GPL) is available at
 * http://www.gnu.org/copyleft/gpl.html.
 *
 * Contact help@fusionticket.com if any conditions of this licencing isn't
 * clear to you.
 */

// het random nr. aanmaken en gecodeerd opslaan in php sessie
define('ft_check','shop');
//require_once('includes/config/defines.php');
//require_once('includes/classes/basics.php');
error_reporting(0);

session_name('ShopSession');
session_start();

if (isset($_POST['check'])) {
  writelog(print_r($_SESSION['_NoSpam'],true));
  writelog(print_r($_POST,true));
  $check = $_SESSION['_NoSpam'][clean($_POST['name'])] == md5(strtoupper ($_POST['check']));
  echo json_encode($check);
  exit;
}

$randomnr = '';

// captcha plaatje met nummer maken - afmetingen kun je aanpassen gebruikte font

$im = imagecreatetruecolor(100, 46);

// Kleurenbepaling

$grey = imagecolorallocate($im, 198, 198, 198);
$black = imagecolorallocate($im, 0, 0, 0);

// zwarte rechthoek tekenen - afmetingen kun je aanpassen aan verschillende fonts

imagefilledrectangle($im, 0, 0, 100, 46, imagecolorallocate($im, rand(120,255), rand(120,255), rand(120,255)));

 for ($i = 1; $i < 10; $i++) {
    imagefilledrectangle($im,rand(0,50),rand(0,23),rand(50,100),rand(23,46),imagecolorallocate($im,mt_rand(120,255),mt_rand(120,255),mt_rand(120,255)));
    imagefilledellipse($im,rand(0,100),rand(0,60),rand(25,50),rand(25,50),imagecolorallocate($im,mt_rand(120,255),mt_rand(120,255),mt_rand(120,255)));
 }
// hier - font.ttf' vervangen met de locatie van je eigen font bestand
$font = 'includes/fonts/Gibberish.ttf';
$text = '2346789ABCEFGHKNPRT';
// schaduw toevoegen
// voorkomen dat afbeelding ge-cached wordt
 for ($i = 0; $i < 2500; $i++) {
  	$color_pixel  = imagecolorallocatealpha ($im, mt_rand(200, 255), mt_rand(200, 255), mt_rand(200, 255),64);
		ImageSetPixel($im, rand(0, 100), rand(0, 46), $color_pixel);
 }
 $white = imagecolorallocate($im,0,0,0); //mt_rand(010,120), mt_rand(010,120), mt_rand(010,120)); //

  for ($i = 0; $i < 5; $i++) {
   $char = substr($text,rand(0,strlen($text)-1),1);
   $randomnr .= $char;
   $angle = rand(-15,15);
   $y = rand(-15,5);
   if (function_exists('imagettftext')) {
     imagettftext($im, 16, $angle, 7+($i*19), 34+$y, $grey, $font, $char);
     imagettftext($im, 16, $angle, 5+($i*19), 36+$y, $white, $font, $char);
   } else {
     imagestring($im, 5,  7+($i*19), 14+$y, $char, $grey);
     imagestring($im, 5,  5+($i*19), 16+$y, $char, $white);
   }
 }

if (!isset($_GET['name'])) {$_GET['name'] ='RandomNr';}
$_SESSION['_NoSpam'][clean($_GET['name'])] = md5($randomnr);

header("Expires: Wed, 1 Jan 1997 00:00:00 GMT");
header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
header("Cache-Control: no-store, no-cache, must-revalidate");
header("Cache-Control: post-check=0, pre-check=0", false);
header("Pragma: no-cache");

// plaatje verzenden naar browser

header ("Content-type: image/gif");
imagegif($im);
imagedestroy($im);
// =====================================
function strip_tags_in_big_string($textstring){
  $safetext = '';
  while (strlen($textstring) != 0)
  {
    $temptext = strip_tags(substr($textstring,0,1024));
    $safetext .= $temptext;
    $textstring = substr_replace($textstring,'',0,1024);
  }
  return $safetext;
}

function wp_entities($string, $encode = 1){
  $a = (int) $encode;

  if($a == 1) {
    $original = array("'"=>"&%39;",  "\""=> "&%34;" ,"("=>"&#40;"    ,")"=> "&#41;", "`" =>"&apos;" );//,"#"=>"&%35;"
    return strtr( $string, $original);
  } else {
    $original = array("'"   ,"\""   ,"#"    ,"("    ,")", "`"  );
    $entities = array("&%39;","&%34;","&%35;","&#40;","&#41;","&apos;");
    return str_replace($entities, $original, $string);
  }
}

function clean($string, $type='ALL') {

  switch (strtolower($type)) {
    case 'revert':
      return  htmlspecialchars_decode(wp_entities($string,0),ENT_QUOTES );
      break;
    case 'all'  : $string = strip_tags_in_big_string ($string);
    case 'strip': $string = html_entity_decode($string, ENT_QUOTES,'UTF-8');
    case 'html' : $string = htmlentities($string, ENT_QUOTES, "UTF-8");
    case 'htmlz' : $string = wp_entities($string);

  }
  return $string;
}
function writeLog($what, $where = FT_DEBUG){
  Global $_SHOP;
  if ($where < 0) { return; }

  $logname = 'error';
  if ($where == FT_DEBUG) {
    $logname = 'debug';
  }
  if (!isset($_SHOP->hasloged[$where])){
    $what = date('d.m.Y H:i ') ."--------------------------------\n". $what;
    $_SHOP->hasloged[$where] =1;
  }
  $h = fopen('includes/temp/' . $logname.'.'.date('Y-m-d') . '.log', 'a');
  if ($h) {
    fwrite($h,utf8_encode($what . "\n"));
    fclose($h);
  }
}

?>